### PR TITLE
Plumb rendering options to kernels

### PR DESCRIPTION
**Pull request by fmz**

### CODE REVIEW

**Comment by fmz** [comment]
This change is larger than it strictly needs to be. Please forgive the large format changes

**Comment by fmz** [comment]
@Kauhentus @austin-funk Lmk what you think

**Comment by Kauhentus** [comment]
Looks awesome! I'll merge it into restir-dev having been shown the ropes 🤞

**Comment by Kauhentus** [comment]
tragedy has occured let me try again....

**Comment by Kauhentus** [comment]
ok so I ran restir-ref2 and it was working very nicely so I merged it and resolved the few conflicts. but then when I went to finally push *somehow* the new psnr changes were also in restir-ref2 and got merged in too. I must apologize here ... this was probably not what you intended ... 😔
